packages/flutter/lib/src/material/user_accounts_drawer_header.dart

```diff
@@ -60,75 +62,119 @@ class _AccountPictures extends StatelessWidget {
   }
 }
 
-class _AccountDetails extends StatelessWidget {
+class _AccountDetails extends StatefulWidget {
   const _AccountDetails({
-    Key key,
     @required this.accountName,
     @required this.accountEmail,
     this.onTap,
     this.isOpen,
-  }) : super(key: key);
+  });
 
   final Widget accountName;
   final Widget accountEmail;
   final VoidCallback onTap;
   final bool isOpen;
 
+  @override
+  _AccountDetailsState createState() => new _AccountDetailsState();
+}
+
+class _AccountDetailsState extends State<_AccountDetails> with SingleTickerProviderStateMixin {
+  Animation<double> _animation;
+  AnimationController _controller;
+
+  @override
+  void initState () {
+    super.initState();
+
+    _controller = new AnimationController(
+      value: widget.isOpen ? 1.0 : 0.0,
+      duration: const Duration(milliseconds: 200),
+      vsync: this,
+    );
+    _animation = new CurvedAnimation(
+        parent: _controller,
+        curve: Curves.easeInOut,
+      )
+      ..addListener(() {
+        setState(() {});
```

Comment:
always put a comment inside the block of a setState() if it's otherwise empty, explaining what it is that the build method depends on that has changed